WordPress/Classes/ViewRelated/Post/PostListViewController.m

```diff
@@ -0,0 +1,1319 @@
+@property (nonatomic, weak) UITableView *tableView;
+@property (nonatomic, weak) UIRefreshControl *refreshControl;
+@property (nonatomic, strong) WPTableViewHandler *tableViewHandler;
+@property (nonatomic, strong) WPContentSyncHelper *syncHelper;
+@property (nonatomic, strong) PostCardTableViewCell *textCellForLayout;
+@property (nonatomic, strong) PostCardTableViewCell *imageCellForLayout;
+@property (nonatomic, strong) PostCardTableViewCell *thumbCellForLayout;
+@property (nonatomic, strong) WPNoResultsView *noResultsView;
+@property (nonatomic, strong) PostListFooterView *postListFooterView;
+@property (nonatomic, weak) IBOutlet NavBarTitleDropdownButton *filterButton;
+@property (nonatomic, weak) IBOutlet UIView *rightBarButtonView;
+@property (nonatomic, weak) IBOutlet UIButton *searchButton;
+@property (nonatomic, weak) IBOutlet UIButton *addButton;
+@property (nonatomic, weak) IBOutlet UIView *searchWrapperView; // Used on iPhone for presenting the search bar.
+@property (nonatomic, weak) IBOutlet UIView *authorsFilterView; // Search lives here on iPad
+@property (nonatomic, weak) IBOutlet UISegmentedControl *authorsFilter;
+@property (nonatomic, weak) IBOutlet NSLayoutConstraint *authorsFilterViewHeightConstraint;
+@property (nonatomic, weak) IBOutlet NSLayoutConstraint *searchWrapperViewHeightConstraint;
+@property (nonatomic, strong) WPSearchController *searchController; // Stand-in for UISearchController
+@property (nonatomic, strong) UIPopoverController *postFilterPopoverController;
+@property (nonatomic, strong) NSArray *postListFilters;
+@property (nonatomic, strong) NSMutableArray *recentlyTrashedPostIDs; // IDs of trashed posts. Cleared on refresh or when filter changes.
+
+@end
+
+@implementation PostListViewController
+
+#pragma mark - Lifecycle Methods
+
++ (instancetype)controllerWithBlog:(Blog *)blog
+{
+    UIStoryboard *storyboard = [UIStoryboard storyboardWithName:@"Posts" bundle:[NSBundle mainBundle]];
+    PostListViewController *controller = [storyboard instantiateViewControllerWithIdentifier:@"PostListViewController"];
+    controller.blog = blog;
+    controller.restorationClass = [self class];
+    return controller;
+}
+
++ (UIViewController *)viewControllerWithRestorationIdentifierPath:(NSArray *)identifierComponents coder:(NSCoder *)coder
+{
+    NSString *blogID = [coder decodeObjectForKey:PostsViewControllerRestorationKey];
+    if (!blogID) {
+        return nil;
+    }
+
+    NSManagedObjectContext *context = [[ContextManager sharedInstance] mainContext];
+    NSManagedObjectID *objectID = [context.persistentStoreCoordinator managedObjectIDForURIRepresentation:[NSURL URLWithString:blogID]];
+    if (!objectID) {
+        return nil;
+    }
+
+    NSError *error = nil;
+    Blog *restoredBlog = (Blog *)[context existingObjectWithID:objectID error:&error];
+    if (error || !restoredBlog) {
+        return nil;
+    }
+
+    return [self controllerWithBlog:restoredBlog];
```

Comment:
It seems to restore everything correctly (including filters), except scroll offset. I'm not sure if that's expected or much of a problem